src/hooks/useFetch.js

```diff
@@ -0,0 +1,118 @@
+import 'abortcontroller-polyfill/dist/abortcontroller-polyfill-only';
+import { useState, useEffect } from 'react';
+import { getAuthHeaders } from 'shared/services/auth/auth';
+import { getErrorMessage } from 'shared/services/api/api';
+
+export const headers = {
+  ...getAuthHeaders(),
+  'Content-Type': 'application/json',
+  Accept: 'application/json',
+};
+
+/**
+ * Custom hook useFetch
+ *
+ * Will take a URL and an optional object of parameters and use those to construct a request URL
+ * with, call fetch and return the response.
+ *
+ * @param {Object} params - key/value
+ * @returns {FetchResponse}
+ */
+export default () => {
+  const [isLoading, setLoading] = useState(false);
+  const [data, setData] = useState();
+  const [error, setError] = useState(false);
+  const [isSuccess, setSuccess] = useState();
+
+  const controller = new AbortController();
+  const { signal } = controller;
+
+  useEffect(() => () => {
+    controller.abort();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const get = async (url, params) => {
+    setLoading(true);
+
+    const queryParams = Object.entries(params || {})
+      .filter(([, value]) => Boolean(value))
+      .reduce((acc, [key, value]) => [...acc, `${key}=${value}`], [])
+      .join('&');
+
+    const requestURL = [url, queryParams].filter(Boolean).join('?');
+
+    try {
+      const response = await fetch(requestURL, {
+        headers,
+        signal,
+      });
+      /* istanbul ignore else */
+      if (response.ok === false) {
+        throw response;
+      }
+
+      const JSONResponse = await response.json();
+
+      setData(JSONResponse);
+    } catch (e) {
```

Comment:
Personally I prefer not to use 1 letter variables, error (or event) is much better 'greppable' then 'e', my 2 cents, feel free to ignore!

Reply:
Good one; I'll take that into account for future PRs